source/Calamari.Shared/Integration/Scripting/ScriptEngine.cs

```diff
@@ -26,7 +27,25 @@ public abstract class ScriptEngine : IScriptEngine
 
                 try
                 {
-                    result = commandLineRunner.Execute(execution.CommandLineInvocation);
+                    if (execution.CommandLineInvocation.Isolate)
+                    {
+                        using (var syncMutex = new Mutex(true, "PythonSynchronizeInitialization",
```

Comment:
The name is very specific to python but in a general implementation

Reply:
what should we call it?

Reply:
fixed